app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
@iNecas, basically sounds good. Let me tell you my assumptions:

1. Each task knows how to serialize/deserialize its own objects.
2. Dynflow framework doesn't care how the recipient task will handle the output of the sender task.
3. Dynflow needs to supply a stub for the output object in planning phase in order to understand the dependency between the tasks.
4. Dynflow doesn't want to limit the interface between tasks to specific object types (like hash, array and primitives).

Am I missing something?